server/game/cards/15-DotE/HothoHumpback.js

```diff
@@ -7,12 +7,18 @@ class HothoHumpback extends DrawCard {
             title: 'Draw card and check reserve',
             phase: 'challenge',
             message: '{player} uses {source} to have each player draw a card and check for reserve',
-            gameAction: GameActions.simultaneously(() => [
-                ...this.game.getPlayersInFirstPlayerOrder().map(player =>
-                    GameActions.drawCards({ player, amount: 1 })
-                ),
-                GameActions.checkReserve()
-            ]),
+            handler: context => {
+                this.game.resolveGameAction(
+                    GameActions.simultaneously(() => [
+                        ...this.game.getPlayersInFirstPlayerOrder().map(player =>
+                            GameActions.drawCards({ player, amount: 1 })
+                        )
+                    ]).then({

```

Comment:
This *might* be worth merging, but it is not correct (otherwise I would have implemented this way in the first place). 

Consider if either played has a restriction on the number of cards they can draw (e.g. High Sparrow), and cannot draw the card for Hotho. Since the pre-then condition was unsuccessful - either one or both players did not draw a card - then the post-then check reserve will not resolve. But Hotho's card is phrased in a way such that reserve should always be checked, regardless of the success of the draw.

Reply:
Thanks for your review, you´ve made a really good point i didn´t think about. I made according changes in the latest commit and added a test to check for the edge case of a draw restriction in play.

Reply:
Thanks! Theoretically the original code should have worked, but pipeline ordering is hard to understand. Adding the test is a big improvement, and will let us refactor this to something better later.